native/cocos/renderer/gfx-metal/MTLCommandBuffer.mm

```diff
@@ -294,12 +294,19 @@ of this software and associated engine source code (the "Software"), a limited,
         mu::clearRenderArea(_mtlDevice, _renderEncoder.getMTLEncoder(), renderPass, renderArea, colors, depth, stencil);
     }
 
+    const auto &targetSize = ccMtlRenderPass->getRenderTargetSizes()[0];
```

Comment:
In non performance critical path, `at` is safer than `operator[]`. Memory access violation is hard to debug caused by `operator[]`, which does not check iterator range.

Reply:
`RenderTargetSizes` might be zero. Some depth-prepass only bind depth-stencil texture.

Reply:
> `at` is safer than `operator[]`

Yes, `at` would be better.

> `RenderTargetSizes` might be zero.

I have checked `RenderTargetSizes`. When renderpass adding a depth stencil attachment, it will check whether `RenderTargetSizes` is empty, and if it is, it will add a targetSize.(Although I think it's a little bit weird.) So a depth-only render pass will also have a size record.

Reply:
Yes. We might refactor targetSize later.